backend/layers/common/entities.py

```diff
@@ -0,0 +1,65 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import List
+
+
+@dataclass
+class DatasetStatus:
+    status: str  # TODO: use an enum
+
+
+@dataclass
+class DatasetArtifact:
+    type: str
+    uri: str
+
+
+@dataclass
+class DatasetMetadata:
+    organism: str
+    tissue: str
+    assay: str
+    disease: str
+    sex: str
+    self_reported_ethnicity: str
+    development_stage: str
+    cell_type: str
+    cell_count: int
+
+
+@dataclass
+class Dataset:
+    id: str
+    status: DatasetStatus
+    metadata: DatasetMetadata
+    artifacts: List[DatasetArtifact]
+
+
+@dataclass
+class CollectionMetadata:
+    name: str
+    description: str
+    owner: str
+    contact_name: str
+    contact_email: str
+    links: List[dict]  # TODO: use a dataclass
+
+
+@dataclass
+class Collection:
```

Comment:
We probably do. I was initially thinking to leave tombstones entirely within the persistence layer (e.g., `get_collection` could raise an exception in case the selected collection is tombstoned), but it might make more sense to leave that logic up to the business layer. Any preference?